### PR TITLE
[FSR] 64213 - Review page navigation

### PR DESCRIPTION
**Pull request by Dr-Pongo**
## Summary
This update includes change so we have "Edit section" buttons on the final review page at the top of each chapter section, and includes navigation additions to chapter heads and tails to return to the review page if they used the new navigation. 

This change included updating the following pages to fully custom components to handle info alerts, and custom continue/back page navigations:

- Veteran information
- Dependent count 
- Veteran employment question 
- Household Expense explainer
- Monetary Assets checklist
- Resolution explainer
- Resolution comments 
- Bankruptcy Question
- Bankruptcy details

Included custom review page pages for the following to include "edit section" headers:

- Veteran information
- Veteran employment
- Expenses explainer
- Real estate 
- Resolution explainer
- Bankruptcy Question

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#64213

## Testing done

- Local testing completed. 
- Updated e2e cypress tests to account for new custom pages

## Screenshots
_Screenshot coming soon_
_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?
*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/financial-status-report/components/alerts/ReviewPageNavigationAlert.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      }),
+    );
+  };
+
+  return (
+    <va-alert
+      class="vads-u-margin-y--1"
+      status="info"
+      data-testid="review-page-navigation-alert"
+      uswds
```
Comment (josephrlee) [inline src/applications/financial-status-report/components/alerts/ReviewPageNavigationAlert.jsx:23]: Hey @Dr-Pongo, can you make a couple small UI tweaks to the alert? 

Can you add the `background-only` prop and change class to `vads-u-margin-bottom--2`?

File: src/applications/financial-status-report/components/shared/ReviewPageHeader.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * @param {String} [title] - Title of section.
+ * @param {Function} goToPath - goToPath passed in from CustomPage/CustomPageReview props.
+ * @return {React Component}
+ */
+const ReviewPageHeader = ({ title, goToPath }) => {
+  return (
+    <div className="form-review-panel-page" key="income-edit-navigation">
+      <div className="form-review-panel-page-header-row">
+        <h4 className="form-review-panel-page-header vads-u-font-size--h5">
```
Comment (josephrlee) [inline src/applications/financial-status-report/components/shared/ReviewPageHeader.jsx:16]: Can you remove the `vads-u-font-size--h5`? I wanted the section titles to look a little bigger than the individual field titles to differentiate the visual hierarchy.

File: src/applications/financial-status-report/components/alerts/ReviewPageNavigationAlert.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    >
+      <h4 slot="headline">Editing {title}</h4>
+      <p className="vads-u-font-size--base vads-u-font-family--sans">
+        You are currently editing the {title} section. Complete the entire
```
Comment (amponce) [inline src/applications/financial-status-report/components/alerts/ReviewPageNavigationAlert.jsx:27]: Need to convert to variable: {title}

**Comment by Dr-Pongo** [comment]
This PR has been split into two other PRs so it's easier to reveiw and the changes are more manageable.

- Changing pages to CustomPages to allow flow
  - #26272
- Adding review functionality behind feature flag
  - #26509

**Pull request closed without merging**